packages/picker/test/index.ts

```diff
@@ -1412,6 +1412,41 @@ export function runPickerTests(): void {
                 'has focus ring?'
             ).to.be.false;
         });
+        it('does not close on document scroll', async () => {
+            const el = await fixture(html`
+                <div style="height: 200vh; padding: 50vh 0;">
+                    <sp-picker label="Select an option" placement="right">
+                        <sp-menu-item value="option-1">Option 1</sp-menu-item>
+                        <sp-menu-item value="option-2">Option 2</sp-menu-item>
+                        <sp-menu-item value="option-3">Option 3</sp-menu-item>
+                    </sp-picker>
+                </div>
+            `);
+
+            const picker = el.querySelector('sp-picker') as Picker;
+            await elementUpdated(picker);
+            await nextFrame();
+            await nextFrame();
```

Comment:
I'm curious what purpose do these duplicate awaits serve? Is there an event or specific state we're waiting on that can be queried for instead to ensure we don't add a new test that might be flaky as a result of an arbitrary wait time?

Reply:
This is a way we use this API to wait for multiple rendering cycles to ensure the DOM has fully updated and any potential asynchronous operations like overlay positioning to have completed.

Reply:
so i have learned this can cause flakiness, did you try elementUpdated(picker)?

Reply:
Right I understand what purpose it serves but we also have a ton of flaky tests as a result of this kind of pattern which is not awaiting a specific action but an arbitary amount of time.